File: src/main/java/configuration/IndexController.java
package configuration;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.servlet.ModelAndView;
import twirl.Redirect;
import twirl.TwirlView;

/**
 * User: 67726e
 */
@Controller
@RequestMapping(value = "/")
public class IndexController {
	@RequestMapping(value = "/", method = RequestMethod.GET)
	public ModelAndView getIndex() {
		return new ModelAndView("index");
	}

	@RequestMapping(value = "/test.html", method = RequestMethod.GET)
	public TwirlView getHtml() {
		return new TwirlView(html.index.apply());
	}

	@RequestMapping(value = "/test.js", method = RequestMethod.GET)
	public TwirlView getJs() {
		return new TwirlView(js.application.apply());
	}

	@RequestMapping(value = "/test.xml", method = RequestMethod.GET)
	public TwirlView getXml() {
		return new TwirlView(xml.test.apply());
	}

	@RequestMapping(value = "/test.txt", method = RequestMethod.GET)
	public TwirlView getText() {
		return new TwirlView(txt.test.apply());
	}

	@RequestMapping(value = "/redirect", method = RequestMethod.GET)
	public TwirlView getRedirect() {
		return new TwirlView(new Redirect("/test.html"));
	}
}

File: src/test/scala/controller/IndexController.scala
package controller

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.{RequestMapping, RequestMethod}
import us.hexcoder.twirl.view.TwirlView
import us.hexcoder.twirl.{Redirect, TwirlView}

/**
 * User: 67726e
 */
@Controller
@RequestMapping(value = Array("/"))
class IndexController {
//	@RequestMapping(value = Array("/"), method = Array(RequestMethod.GET))
//	def index():TwirlView = {
//		new TwirlView()
//	}

	@RequestMapping(value = Array("/test.html"), method = Array(RequestMethod.GET))
	def getHtml():TwirlView = {
        TwirlView.ok(html.index())
    }

	@RequestMapping(value = Array("/test.xml"), method = Array(RequestMethod.GET))
	def getXml():TwirlView = {
		TwirlView.ok(xml.test())
	}

	@RequestMapping(value = Array("/test.txt"), method = Array(RequestMethod.GET))
	def getTxt():TwirlView = {
        TwirlView.ok(txt.test())
	}

	@RequestMapping(value = Array("/test.js"), method = Array(RequestMethod.GET))
	def getJs():TwirlView = {
        TwirlView.ok(js.application())
	}

	@RequestMapping(value = Array("/redirect"), method = Array(RequestMethod.GET))
	def redirect():TwirlView = {
        TwirlView.redirect("/test.html")
	}
}
